### PR TITLE
MIL-57 Docker Compose

### PR DESCRIPTION
**Pull request by matizaj**
Add nginx as a proxy server, 
add Dockerfile for frontend apllication, 
add docker-compose file

### CODE REVIEW

File: frontend/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+../.idea
+dist
```
Comment (pawelkonior) [inline frontend/.dockerignore:3]: new line at the end of file?

File: frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM node:16.13.1-alpine
+WORKDIR /app
+COPY package.json package-lock.json ./
+RUN npm install
+COPY . .
+CMD ["npm", "run", "build"]
```
Comment (pawelkonior) [inline frontend/Dockerfile:6]: new line at the end of file?

File: proxy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+FROM nginx
+COPY ./default.conf /etc/nginx/conf.d/default.conf
```
Comment (pawelkonior) [inline proxy/Dockerfile:2]: new line at the end of file?

File: proxy/default.conf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+server {
+    listen 80;
+
+    location / {
+        proxy_pass http://ui;
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $remote_addr;
+    }
+
+}
```
Comment (pawelkonior) [inline proxy/default.conf:14]: new line at the end of file?

File: frontend/Dockerfile (hunk 2 above)
Comment (pawelkonior) [reply frontend/Dockerfile:6]: CMD ["npm", "run", "dev"]

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  proxy:
+    build:
+      context: ./proxy
+    ports:
+      - "8083:80"
+  ui:
+    build:
+      context: ./frontend
+    ports:
+      - "3000:3000"
```
Comment (pawelkonior) [inline docker-compose.yml:11]: - "3000:8080"

File: frontend/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FROM node:16.13.1-alpine
```
Comment (pawelkonior) [inline frontend/Dockerfile:1]: 16.13.2 (actual LTS)

**Review by pawelkonior — APPROVED** [review]
Awesome!

File: frontend/.dockerignore (hunk 1 above)
Comment (matizaj) [reply frontend/.dockerignore:3]: ok

File: frontend/Dockerfile (hunk 2 above)
Comment (matizaj) [reply frontend/Dockerfile:6]: done

File: proxy/Dockerfile (hunk 3 above)
Comment (matizaj) [reply proxy/Dockerfile:2]: fix

File: docker-compose.yml (hunk 5 above)
Comment (matizaj) [reply docker-compose.yml:11]: changed

File: frontend/Dockerfile (hunk 6 above)
Comment (matizaj) [reply frontend/Dockerfile:1]: changed

**Pull request closed without merging**